docs/reduction_toolbox.rst

```diff
@@ -428,6 +432,112 @@ which should be removed rather than propagated:
+
+The example begins by creating small MEF "science" image and a sample flat:
+
+    >>> import numpy as np
+    >>>
+    >>> from astropy.utils.misc import NumpyRNGContext
+    >>> from astropy.io import fits
+    >>> from astropy.nddata import CCDData
+    >>>
+    >>> from ccdproc import flat_correct
+    >>>
+    >>>
+    >>> def make_sample_mef(science_name, flat_name):
+    ...     """
+    ...     Make a multi-extension FITS image with random data
+    ...     and a MEF flat.
+    ...
+    ...     Parameters
+    ...     ----------
+    ...
+    ...     science_name : str
+    ...         Name of the science image created by this function.
+    ...
+    ...     flat_name : str
+    ...         Name of the flat image created by this function.
+    ...     """
+    ...     with NumpyRNGContext(1234):
+    ...         number_of_image_extensions = 3
+    ...         science_image = [fits.PrimaryHDU()]
+    ...         flat_image = [fits.PrimaryHDU()]
+    ...         for _ in range(number_of_image_extensions):
+    ...             # Simulate a cloudy night, average pixel
+    ...             # value of 100 with a read_noise of 1 electron.
+    ...             # Image size is 150 × 150.
+    ...             data = np.random.normal(100., 1.0, [150, 150])
+    ...             hdu = fits.ImageHDU(data=data)
+    ...             # Make a header that is at least somewhat realistic
+    ...             hdu.header['unit'] = 'electron'
+    ...             hdu.header['object'] = 'clouds'
+    ...             hdu.header['exptime'] = 30.0
+    ...             hdu.header['date-obs'] = '1928-07-23T21:03:27'
+    ...             hdu.header['filter'] = 'B'
+    ...             hdu.header['imagetyp'] = 'LIGHT'
+    ...             science_image.append(hdu)
+    ...
+    ...             # Make a perfect flat
+    ...             flat = np.ones_like(data)
+    ...             flat_hdu = fits.ImageHDU(data=flat)
+    ...             flat_hdu.header['unit'] = 'electron'
+    ...             flat_hdu.header['filter'] = 'B'
+    ...             flat_hdu.header['imagetyp'] = 'FLAT'
+    ...             flat_hdu.header['date-obs'] = '1928-07-23T21:03:27'
+    ...             flat_image.append(flat_hdu)
+    ...
+    ...     science_image = fits.HDUList(science_image)
+    ...     science_image.writeto(science_mef)
+    ...
+    ...     flat_image = fits.HDUList(flat_image)
+    ...     flat_image.writeto(flat_mef)
+    >>>
+    >>> science_mef = 'science.fits'
+    >>> flat_mef = 'flat.fits'
+    >>>
+    >>> make_sample_mef(science_mef, flat_mef)
+    >>>
+    >>> # CALIBRATION EXAMPLE STARTS HERE
+    >>> # (everything above this was to make sample files)
+    >>>
+    >>> # Read our sample images
+    >>> science = fits.open(science_mef)
```

Comment:
This will leave the files open and should really be avoided in official "examples". One could use the double `with` here:

```
with fits.open(science_mef) as science, fits.open(flat_mef) as flat:
```

Reply:
Maybe even use `memmap=False` because we're in the global scope so the file-handle will be kept alive even if the context closes (in a function the dropping of the reference would finally close the file).

Reply:
I added explicit `.close` calls at the end of the example. I agree it is bad to leave them open. I think in this case `memmap=False` is a bad idea because may well copy/paste the code. `with` is probably preferable but added another level of indentation.

Reply:
> I think in this case `memmap=False` is a bad idea because may well copy/paste the code.

The memmap is actually *because* people may copy/paste the code. Closing the memmap'd file just closes the file and doesn't close the memmap if the variables referencing the memmap aren't disposed of.

> `with` is probably preferable but added another level of indentation.

I agree but I think that's a worth-while level of indentation :)